plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -67,14 +83,25 @@ function isBlank(str: string) {
  */
 export class ElasticSearchSearchEngine implements SearchEngine {
   private readonly elasticSearchClient: Client;
+  private readonly highlightOptions: ElasticSearchHighlightOptions;
 
   constructor(
     private readonly elasticSearchClientOptions: ElasticSearchClientOptions,
     private readonly aliasPostfix: string,
     private readonly indexPrefix: string,
     private readonly logger: Logger,
+    highlightOptions?: ElasticSearchHighlightOptions,
   ) {
     this.elasticSearchClient = this.newClient(options => new Client(options));
+    const uuidTag = uuid();
+    this.highlightOptions = {
+      preTag: `<${uuidTag}>`,
```

Comment:
The motivation behind this is that we don't to potentially clash with content that contain HTML. For example, if a TechDocs page contains HTML code examples that include the `<mark>` tag, it would be improperly highlighted in the UI. Using a uuid ensures we will have distinct markers in the content that the frontend can parse and replace with actual `<mark>` elements. We also don't want to format actual HTML here to render directly since that would involve `dangerouslySetInnerHTML` which of course has security concerns